### PR TITLE
Add a way to specify the version of the cluster agent

### PR DESCRIPTION
**Pull request by L3n41c**
What does this PR do?
---------------------

Add a way to specify the version of the cluster agent.
Align the version of the cluster checks runners with the version of the node agents.

Which scenarios this will impact?
-------------------

Cluster agent and cluster checks runners are specific to Kubernetes setups.

Motivation
----------

We need a way to specify the version of the cluster agent to use.
We could enforce to use the same version for the cluster agent and the agent but in any case, they are in two different docker images.

Additional Notes
----------------

### CODE REVIEW

File: datadog/agent/docker.go [hunk 1]
```diff
@@ -44,8 +46,36 @@ func DockerImageTag(e *config.CommonEnvironment) string {
...
 		agentImageTag = agentVersion.String()
 	} else {
-		e.Ctx.Log.Debug("Unable to parse Agent version, using latest", nil)
+		e.Ctx.Log.Debug("Unable to parse agent version, using latest", nil)
 	}
 
 	return agentImageTag
 }
+
+func DockerClusterAgentFullImagePath(e *config.CommonEnvironment, repositoryPath string) string {
```
Comment (vboulineau) [inline datadog/agent/docker.go:55]: We can probably have a single function by passing default values in

File: common/config/environment.go [hunk 2]
```diff
@@ -27,11 +27,13 @@ const (
 	DDInfraOSFamily          = "osFamily"
 
 	// Agent Namespace
-	DDAgentDeployParamName        = "deploy"
-	DDAgentVersionParamName       = "version"
-	DDAgentFullImagePathParamName = "fullImagePath"
-	DDAgentAPIKeyParamName        = "apiKey"
-	DDAgentAPPKeyParamName        = "appKey"
+	DDAgentDeployParamName               = "deploy"
+	DDAgentVersionParamName              = "agentVersion"
```
Comment (vboulineau) [inline common/config/environment.go:31]: We probably don't need to rename, we have `ddagent:version`, having `ddagent:agentVersion` is a bit redundant.

File: common/config/helpers.go [hunk 3]
```diff
@@ -24,11 +24,19 @@ func SetConfigDefaultValue(config auto.ConfigMap, key, value string) {
 	}
 }
 
-func AgentSemverVersion(e *CommonEnvironment) (*semver.Version, error) {
-	version := e.AgentVersion()
+func anyAgentSemverVersion(anyAgentVersion func() string) (*semver.Version, error) {
```
Comment (ogaca-dd) [inline common/config/helpers.go:27]: What is the reason to pass a `func() string` and not the version directly?

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Pull request merged**